CRM/Admin/Form/Setting/Search.php

```diff
@@ -61,19 +61,13 @@ public function buildQuickForm() {
 
     parent::buildQuickForm();
 
-    // @todo remove the following adds in favour of setting via the settings array (above).
-
-    // Autocomplete for Contact Search (quick search etc.)
+    // Option 1 can't be unchecked. @see self::enableOptionOne
     $element = $this->getElement('contact_autocomplete_options');
-    $element->_elements[0]->_flagFrozen = TRUE;
+    $element->_elements[0]->setAttribute('disabled', 'disabled');
```

Comment:
ideally we would also specific required_options in metadata & handle this there (I'm OK with this being out of scope for this but possibly this whole form could be removed - like https://github.com/civicrm/civicrm-core/pull/14425 if we did move that to metadata

Reply:
I'm not sure that the current class structure is such a bad one. Makes sense to me to have all settings form inherit from a parent which does all the heavy lifting but small tweaks and overrides can be done in those classes.

These specific tweaks strike me as too weird to be generalizeable and I'm not entirely sure that "no form classes for settings pages" ought to be our goal, especially if it means "abstracting" every quirky setting behavior into a one-off metadata property.